base/src/main/java/org/aya/tyck/pat/PatTycker.java

```diff
@@ -169,10 +169,12 @@ public record LhsResult(
   }
 
   private LhsResult checkLhs(Pattern.Clause match, Def.Signature signature) {
+    var ctx = exprTycker.localCtx.deriveMap();
+    var resultIsProp = ctx.with((() -> signature.result().computeType(exprTycker.state, ctx) instanceof FormTerm.Prop), signature.param().toArray(new Term.Param[0]));
     var parent = exprTycker.localCtx;
     exprTycker.localCtx = parent.deriveMap();
```

Comment:
`ctx` looks duplicated with `parent.deriveMap()` / `exprTycker.localCtx` after line 175, How about:

```java
var parent = exprTycker.localCtx;
exprTycker.localCtx = parent.deriveMap();
exprTycker.localCtx.with(...);
```

or we let `LocalCtx#with` derive a `LocalCtx` automatically.

Reply:
I like the second solution

Reply:
We might address it in future prs

Reply:
I think this is sufficiently simple for this PR